File: fixtures/lifecycle-identifier-reuse-and-rename/sdk-probe.mjs
// Copyright 2026 Tymofii Pidlisnyi. Apache-2.0 license. See LICENSE.
//
// Records which APIs the npm reference SDK actually exposes for this family.
// Prints one line per probe: supported, or not_supported with the reason. No
// assertions about behaviour, only reachability.
//
//     node fixtures/lifecycle-identifier-reuse-and-rename/sdk-probe.mjs

import fs from 'node:fs'
import path from 'node:path'
import { fileURLToPath } from 'node:url'

// The package does not export ./package.json, so the version is read from the
// installed tree rather than required.
const here = path.dirname(fileURLToPath(import.meta.url))
const version = JSON.parse(
  fs.readFileSync(path.join(here, '..', '..', 'node_modules', 'agent-passport-system', 'package.json'), 'utf8'),
).version
const sdk = await import('agent-passport-system')

const PROBES = [
  ['external identifier binding, issue', 'issueIdentifierBinding'],
  ['external identifier binding, verify', 'verifyIdentifierBinding'],
  ['external identifier controller, resolve at instant', 'resolveIdentifierControllerAt'],
  ['external identifier continuity, verify', 'verifyIdentifierContinuity'],
  ['identifier retention record, issue', 'issueIdentifierRetention'],
  ['namespace claim, create', 'createNamespaceClaim'],
  ['namespace claim, verify', 'verifyNamespaceClaim'],
  ['DID document, create', 'createDIDDocument'],
  ['public key from DID', 'publicKeyFromDID'],
  ['identity rotation log, verify', 'verifyRotationLog'],
  ['unbind event, verify', 'verifyUnbindEvent'],
  ['bound wallet, verify', 'verifyBoundWallet'],
  ['authority chain, verify', 'verifyAuthorityDelegationChain'],
  ['authority delegation, issue root', 'issueAuthorityDelegation'],
  ['RFC 8785 JCS canonical bytes', 'canonicalizeJCS'],
  ['Ed25519 verify', 'verify'],
]

console.log(`agent-passport-system@${version} (npm, TypeScript reference SDK)`)
let notSupported = 0
for (const [label, name] of PROBES) {
  if (typeof sdk[name] === 'function') {
    console.log(`  supported      ${label.padEnd(50)} ${name}`)
  } else {
    notSupported += 1
    console.log(`  not_supported  ${label.padEnd(50)} ${name}: not exported from the package root`)
  }
}
console.log('')
console.log(`${PROBES.length - notSupported}/${PROBES.length} supported. No exported API takes an external identifier`)
console.log('and an instant and answers who controls it, so the dependency, pin, binding, conflict,')
console.log('lapse and continuity-gap steps are implemented in harness.ts, not by the SDK. See README.')

File: fixtures/lifecycle-identifier-reuse-and-rename/sdk-probe.py
#!/usr/bin/env python3
"""Records which APIs the PyPI reference SDK actually exposes for this family.

Prints one line per probe: supported, or not_supported with the reason. No
assertions about behaviour, only reachability.

    python3 fixtures/lifecycle-identifier-reuse-and-rename/sdk-probe.py
"""
from __future__ import annotations

import importlib
import importlib.metadata as metadata

PROBES = [
    ("external identifier binding, issue", "agent_passport", "issue_identifier_binding"),
    ("external identifier binding, verify", "agent_passport", "verify_identifier_binding"),
    ("external identifier controller, resolve at instant", "agent_passport", "resolve_identifier_controller_at"),
    ("external identifier continuity, verify", "agent_passport", "verify_identifier_continuity"),
    ("identifier retention record, issue", "agent_passport", "issue_identifier_retention"),
    ("namespace claim, verify", "agent_passport", "verify_namespace_claim"),
    ("DID document, create", "agent_passport", "create_did_document"),
    ("public key from DID", "agent_passport", "public_key_from_did"),
    ("identity rotation log, verify", "agent_passport", "verify_rotation_log"),
    ("authority chain, verify", "agent_passport.v2.authority_delegation", "verify_authority_delegation_chain"),
    ("authority delegation, issue root", "agent_passport", "issue_authority_delegation"),
    ("RFC 8785 JCS canonical bytes", "agent_passport", "canonicalize_jcs"),
    ("Ed25519 verify", "agent_passport.crypto", "verify"),
]


def main() -> None:
    version = metadata.version("agent-passport-system")
    print(f"agent-passport-system=={version} (PyPI, Python reference SDK)")
    not_supported = 0
    for label, module_name, attribute in PROBES:
        try:
            module = importlib.import_module(module_name)
        except ModuleNotFoundError as error:
            not_supported += 1
            print(f"  not_supported  {label:<50} {module_name}: {error}")
            continue
        if callable(getattr(module, attribute, None)):
            print(f"  supported      {label:<50} {module_name}.{attribute}")
        else:
            not_supported += 1
            print(f"  not_supported  {label:<50} {module_name}.{attribute}: not present in the installed package")
    print("")
    print(f"{len(PROBES) - not_supported}/{len(PROBES)} supported. No exposed API takes an external identifier")
    print("and an instant and answers who controls it, so every identifier step is implemented")
    print("in verify.py, not by the SDK. See README.")


if __name__ == "__main__":
    main()
